### ISSUE TITLE
[Documentation] When should we consider for `ValidateTokenAsync`

### ISSUE THREAD

**Issue by WeihanLi** [AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet#1827]
### Documentation related to component

Currently, we had `ValidateToken` and `ValidateTokenAsync`, for example, when we're using `JwtSecurityTokenHandler` should we prefer using the `ValidateTokenAsync` introduced in 6.16.0.

From what I can see, the `ValidateTokenAsync` could not replace the `ValidateToken` completely, 
the `ValidateToken` method is returning a `ClaimsPrincipal`, while the `ValidateTokenAsync` method is returning a `TokenValidationResult` model which contains a `ClaimsIdentity` but not contains a `ClaimsPrincipal`

### Please check all that apply

- [ ] typo
- [x] documentation doesn't exist
- [ ] documentation needs clarification
- [ ] error(s) in the example
- [ ] needs an example

### Description of the issue

Need some guidance for the `ValidateTokenAsync` method

**Comment by brentschmaltz** [comment]
@WeihanLi token validation returning a ClaimsPrincipal was designed when early tokens (SAML1) could represent two entities.
That idea was replaced believing a token represents a single entity, hence a ClaimsIdentity.
SAML2 and JWT have a single subject.

One of the issues with ValidateToken is the out parameter, which is not available for async calls.

**Comment by AndersAbel** [comment]
> SAML2 and JWT have a single subject.

SAML2 also allows multiple assertions in the same response, which IMHO should result in one principal with multiple identities. As this library only handles SAML2 assertions and not SAML2 responses, anyone implementing support for SAML2 responses (such as the Sustainsys libraries) will have to create a principal with multiple identities.

**Comment by brentschmaltz** [comment]
@AndersAbel I agree with you that at the application layer, ClaimsPrincipal is the right object.
One issue that comes to mind is IsInRole(...), if two identities are involved, who does IsInRole apply to?

**Comment by AndersAbel** [comment]
@brentschmaltz Multiple assertions in the same SAML2 response is a rare thing, I've never heard of it being used, although it is allowed in the spec. For the Web SSO profile, there is a requirement that all identities refer to the same subject, so in that case it would actually work that `IsInRole` checks all identities. But I guess that anyone using multiple identities would probably not use the principal IsInRole anyway.

**Comment by brentschmaltz** [comment]
@AndersAbel with this new API returning a TokenValidationResult, IdentityModel returns a ClaimsIdentity upper layers (such as asp.net) will have to create a ClaimsPrincipal that corresponds to the 'User'.

Does that seem right?

**Comment by AndersAbel** [comment]
@brentschmaltz Yes, that sounds good. For my SAML2P package I can then bundle multiple generated identities from IdentityModel into one ClaimsPrincipal.
